device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -180,7 +169,6 @@ else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509
             this.serverUri = SSL_PREFIX + host + SSL_PORT_SUFFIX;
         }
 
-        MqttAsyncClient mqttAsyncClient = buildMqttAsyncClient(this.serverUri, clientId);
```

Comment:
MqttAsyncClients are single use, so instead of constructing one at MqttIotHubConnection constructor time, we'll create one each time we open the MqttIotHubConnection